pandas/core/sparse/frame.py

```diff
@@ -547,6 +548,31 @@ def xs(self, key, axis=0, copy=False):
     # ----------------------------------------------------------------------
     # Arithmetic-related methods
 
+    def _wrap_dispatched_op(self, result, other, func, axis=None):
+        """
+        Wrap the result of an arithmetic/comparison operation performed
+        via ops.dispatch_to_series in a properly-indexed DataFrame.
+
+        Parameters
+        ----------
+        result : dict[int:Series]
+        other : DataFrame, Series, scalar
+        func : binary function
+        axis : {None, "columns"}
+
+        Returns
+        -------
+        SparseDataFrame
+        """
+
+        fill_value = self._get_op_result_fill_value(other, func, axis)
+
+        res = self._constructor(result, index=self.index,
```

Comment:
why are you passing columns to the constructor here, whereas in DataFrame you are setting?

Reply:
I'm very specifically not changing the existing behavior.